helix-core/src/main/java/org/apache/helix/manager/zk/ZkCacheBaseDataAccessor.java

```diff
@@ -72,6 +72,11 @@
 
   private HelixZkClient _zkClient = null;
 
+  // test only
+  public HelixZkClient getZkClient() {
```

Comment:
This should be introduced. If getZkClient() is really necessary, you can make `_zkClient` as protected.

Reply:
This should *not* be introduced.

Reply:
Protect is not going to work, as the test case live in another package. 

Java has not way to access otherwise private members in the class. In this case, we have to expire the session inside this class. This is a common pattern I did not see addressed in our code base. Unless you guys can identify another pattern, I guess we have to leave it this way.

Reply:
+1, we should not add a public method to a public class just for testing, specially this is a public API.  If you test class is sitting in the same module with this class, protected will work.  Another way is to make it protected, and wrap ZkCacheBaseDataAccessor to a new test class.